lib/src/main/java/io/ably/lib/http/AnySyncHttp.java

```diff
@@ -0,0 +1,69 @@
+package io.ably.lib.http;
+
+import io.ably.lib.types.AblyException;
+import io.ably.lib.types.Callback;
+import io.ably.lib.types.ErrorInfo;
+
+/**
+ * Created by tcard on 28/04/2017.
+ */
+
+public class AnySyncHttp {
+    private final AsyncHttp asyncHttp;
+    private final SyncHttp syncHttp;
+
+    public AnySyncHttp(AsyncHttp asyncHttp, SyncHttp syncHttp) {
+        this.asyncHttp = asyncHttp;
+        this.syncHttp = syncHttp;
+    }
+
+    public static class Request<Result> {
+        private final AnySyncHttp anySyncHttp;
+        private final Execute<Result> execute;
+
+        Request(AnySyncHttp anySyncHttp, Execute<Result> execute) {
+            this.anySyncHttp = anySyncHttp;
+            this.execute = execute;
+        }
+        
+        public Result sync() throws AblyException {
+            final SyncExecuteResult<Result> result = new SyncExecuteResult<>();
```

Comment:
Yes, I was looking at just having `syncResult` and `syncErr` members in the `Result` type, instead of a new type to hold them, but what you have is a bit cleaner.